src/viam/sdk/components/board/board.hpp

```diff
@@ -0,0 +1,162 @@
+    static analog_value from_proto(viam::common::v1::AnalogStatus proto);
+
+    /// @brief Creates a `digital_value` struct from its proto representation.
+    static digital_value from_proto(viam::common::v1::DigitalInterruptStatus proto);
+
+    /// @brief Creates a `power_mode` enum from its proto representation.
+    static power_mode from_proto(viam::component::board::v1::PowerMode proto);
+
+    /// @brief Converts a `status` struct to its proto representation.
+    static viam::common::v1::BoardStatus to_proto(status status);
+
+    /// @brief Converts a `analog_value` struct to its proto representation.
+    static viam::common::v1::AnalogStatus to_proto(analog_value analog_value);
+
+    /// @brief Converts a `digital_value` struct to its proto representation.
+    static viam::common::v1::DigitalInterruptStatus to_proto(digital_value digital_value);
+
+    /// @brief Converts a `power_mode` enum to its proto representation.
+    static viam::component::board::v1::PowerMode to_proto(power_mode power_mode);
+
+    /// @brief Get the status of all of the registered analog readers and digital interrupt readers
+    virtual status get_status() = 0;
+
+    /// @brief Gets the high/low state of the given pin on a board.
+    /// @param pin board pin name
+    /// @return high/low state of the given pin. High = on, low = off
+    virtual bool get_gpio(const std::string& pin) = 0;
+
+    /// @brief Set the gpio high/low state of the given pin on a board
+    /// @param high true if the pin should be set to high (on) or false if it should be low (off)
+    virtual void set_gpio(const std::string& pin, bool high) = 0;
+
+    /// @brief Gets the duty cycle of the given pin on a board.
+    /// @param pin board pin name
+    /// @return duty cycle percentage (0 to 1)
+    virtual double get_pwm_duty_cycle(const std::string& pin) = 0;
+
+    /// @brief Sets the given pin of a board to the given duty cycle.
+    /// @param pin board pin name
+    /// @param duty_cycle_pct duty cycle percentage 0 to 1
+    virtual void set_pwm_duty_cycle(const std::string& pin, double duty_cycle_pct) = 0;
+
+    /// @brief Gets the PWM frequency of the given pin on a board.
+    /// @param pin board pin name
+    virtual uint64_t get_pwm_frequency(const std::string& pin) = 0;
+
+    /// @brief Sets the given pin on a board to the given PWM frequency. 0 will use the board's
+    /// default PWM frequency.
+    /// @param pin board pin name
+    /// @param frequency_hz frequency in hz (0 = use board default frequency)
+    virtual void set_pwm_frequency(const std::string& pin, uint64_t frequency_hz) = 0;
+
+    /// @brief Reads off the current value of an analog reader on a board. Consult your ADC's docs
+    /// or Viam's `Board` docs for more information.
+    /// @param analog_reader_name analog reader to read from
+    virtual analog_value read_analog(const std::string& analog_reader_name) = 0;
```

Comment:
This is awkward. `get_analog` was another contender but then people might accidentally pass in a `pin` rather than the name of the reader.